rotkehlchen/data_import/importers/rotki_events.py

```diff
@@ -0,0 +1,129 @@
+import csv
+from pathlib import Path
+from typing import Any, Dict, List
+from uuid import uuid4
+
+from rotkehlchen.accounting.structures.balance import Balance
+from rotkehlchen.accounting.structures.base import HistoryBaseEntry
+from rotkehlchen.accounting.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.assets.utils import symbol_to_asset_or_token
+from rotkehlchen.constants import ZERO
+from rotkehlchen.data_import.utils import BaseExchangeImporter, UnsupportedCSVEntry
+from rotkehlchen.db.drivers.gevent import DBCursor
+from rotkehlchen.errors.asset import UnknownAsset
+from rotkehlchen.errors.misc import InputError
+from rotkehlchen.errors.serialization import DeserializationError
+from rotkehlchen.serialization.deserialize import (
+    deserialize_asset_amount,
+    deserialize_timestamp_from_date,
+)
+from rotkehlchen.types import Fee, Location, TimestampMS
+
+GENERIC_TYPE_TO_HISTORY_EVENT_TYPE_MAPPINGS = {
+    'Deposit': (HistoryEventType.DEPOSIT, HistoryEventSubType.SPEND),
+    'Withdrawal': (HistoryEventType.WITHDRAWAL, HistoryEventSubType.RECEIVE),
+    'Income': (HistoryEventType.RECEIVE, HistoryEventSubType.RECEIVE),
+    'Loss': (HistoryEventType.SPEND, HistoryEventSubType.NONE),
+    'Staking': (HistoryEventType.STAKING, HistoryEventSubType.REWARD),
+}
+
+
+class RotkiGenericEventsImporter(BaseExchangeImporter):
+    def _consume_rotki_event(
+            self,
+            cursor: DBCursor,
+            csv_row: Dict[str, Any],
+            sequence_index: int,
+            timestamp_format: str = '%Y-%m-%d %H:%M:%S',
+    ) -> str:
+        """Consume Rotki generic events import CSV file.
+        Returns the event identifier
+        May raise:
+        - UnsupportedCSVEntry if an unknown type is encountered.
+        - DeserializationError
+        - UnknownAsset
+        - KeyError
+        """
+        identifier = f'rotki_events_{uuid4().hex}'
+        try:
+            event_type, event_subtype = GENERIC_TYPE_TO_HISTORY_EVENT_TYPE_MAPPINGS[csv_row['Type']]  # noqa: E501
+        except KeyError as e:
+            raise UnsupportedCSVEntry(f'Unsupported entry {csv_row["Type"]}. Data: {csv_row}') from e  # noqa: E501
+        try:
+            location = Location.deserialize(csv_row['Location'])
+        except DeserializationError:
+            location = Location.EXTERNAL
```

Comment:
The whole reason of the generic events import is to allow users to import data from any source provided it can be converted to our import format.

Because of this, we most definitely won't support some locations provided which is why `Location.EXTERNAL` is used in such instance.